### ISSUE TITLE
Need an option to delay rebuilding

### ISSUE THREAD

**Issue by iapyeh** [readthedocs/readthedocs.org#2593]
## Details
I think it is a good feature when a Github repository has new head, the project in the RTD got updated immediately. The problem is that if I commit several times in a short interval, the RTD would lock this project for 5 minutes. 

This is a situation encountered by the GGeditor. GGeditor is a Google Docs add-on. It converts Google Docs document into reStructuredText and commits it to a Github repository. Due to the limitation of the API which is utilized for committing , if a document contains 10 images, the head of repository would be updated for 11 times in a short interval.

## Expected Result

My suggestion is to have an option in the conf.py which would postpone the trigger of rebuilding for a while when a new head of repository is detected. For example, 60 seconds, and the timer would be reset if there is another new head been detected within 60 seconds. It would make sure rebuilding is been triggered only once after the last image-uploading has completed.

With this option, the RTD would also save resources of rebuilding.

## References

- [GGeditor manual in the RTD](http://ggeditor.readthedocs.io/en/latest/)

- [GGeditor add-on in the Google Add-on Store](https://chrome.google.com/webstore/detail/ggeditor/piedgdbcihbejidgkpabjhppneghbcnp)

**Comment by stsewd** [comment]
Maybe https://github.com/rtfd/readthedocs.org/issues/871 is a better solution?

**Comment by humitos** [comment]
More than a user option I think this should be an automatic protection from Read the Docs to avoid people triggering multiple (more than X) builds in the same minute.

We should something similar than `skip=True` at 

https://github.com/rtfd/readthedocs.org/blob/4fa2746040aee0aafcdb09e0e5674dcb42ea9809/readthedocs/core/utils/__init__.py#L96-L101

We need to consider:

* just skipping the trigger of the build could be confusing for the user
* we could check the queue to see if there is already a build queued, remove it from the queue, and add the new one with the new commit (kind of complicated)
* create a Build object but do not trigger it and mark it as FAILED with the correct error message (it will produce to the first build triggered to build and the last --which I think it's the most important-- to not build)

**Comment by stsewd** [comment]
What travis does when receiving multiple webhooks from the same PR is block the others and just build the last one.

**Comment by humitos** [comment]
I think this is solved in some way by the `DEDUPLICATE_BUILDS` feature flag that we recently added.

**Comment by stsewd** [comment]
I think we can close this in favor of https://github.com/readthedocs/readthedocs.org/issues/7660 and https://github.com/readthedocs/readthedocs.org/issues/871 and the already implemented feature of deduplicate builds.
